hazelcast/src/main/java/com/hazelcast/jet/impl/submitjob/memberside/JobMetaDataParameterObject.java

```diff
@@ -58,6 +60,24 @@ public void setSessionId(UUID sessionId) {
         this.sessionId = sessionId;
     }
 
+    public boolean isDirectJobExecution() {
+        return directJobExecution;
+    }
+
+    public void setDirectJobExecution(boolean directJobExecution) {
+        this.directJobExecution = directJobExecution;
+    }
+
+    public void setDeleteJarAfterExecution(boolean deleteJarAfterExecution) {
+        this.deleteJarAfterExecution = deleteJarAfterExecution;
+    }
+
+    public void afterExecution() {
+        if (deleteJarAfterExecution) {
+            deleteJar();
+        }
+    }
```

Comment:
The Jar deletion should go into `JobUploadStatus` where all the temporary jar file handling resides.

Reply:
When all the parts are complete the JobMetaDataParameterObject is removed from the JobUploadStatus. Otherwise while we are executing the jar, the expiration timer may intervene. 
Also I want all the cleanup logic to be in JobMetaDataParameterObject

Reply:
The Parameter object is just a DTO to avoid many constructor parameters.
I think that in general having the logic which creates a resource and cleans it up in a single place makes most sense.

Reply:
Moved the deletion to JobUploadStatus. Deleted deleteJarAfterExecution field
Rev : bb5b1fc1681799745afce0373d490a9846cdbd70